FreeTubeApp/FreeTube
# Make Caption Settings Persistent

---
Made caption settings persistent.
---

**Pull Request Type**

**Related issue**
#989 

**Description**
Make the caption settings persistent across new videos and closing/re-opening the application by:
* Saving caption settings to db when the caption settings modal is closed
* Retrieving the saved caption settings each time a video is opened

**Testing**
1. Open a video with captions
2. Modify caption settings
3. Open a new video with captions and turn captions on, notice caption settings persist
4. Close the application and reopen a video, notice caption settings persist 

**Desktop (please complete the following information):**
 - OS: Mac OS
 - OS Version: 10.15.7
 - FreeTube version: Latest Development Branch

**Additional context**
At the time of creating this PR, issue #1268 may make testing difficult. Please check the comment I left on that issue if caption settings do not appear in video player.
